pkg/pod_lister/byte_order_test.go

```diff
@@ -0,0 +1,21 @@
+package pod_lister
+
+import (
+	"runtime"
+	"encoding/binary"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Byte order test", func() {
+	It("Byte order test", func() {
+		order := determineHostByteOrder()
+		switch runtime.GOARCH {
+			case "amd64":
+				Expect(order).To(Equal(binary.LittleEndian))
+			case "s390x":
```

Comment:
just out of curiosity, any difference between ppc64le and s390x?

Reply:
I am s390x guy :) 

so I didn't count others (e.g ARM, ppc etc)
will update this

Reply:
np, looking forward to a s390x port!

Reply:
> > > > Just wonder why we don't use this as a function to determine byte order instead of the test case.
> 
> has no clue .. as we have a function to decide the order ,a UT is reasonable? or we can use library function to decide so we can avoid both function and UT?

Actually, in my understanding, using unsafe pointer is more valid than GORACH as GOARCH can be overrode by environment variable. (on the other hand, the default value (amd64) might be applied if it is unset)

Anyway, I think it is a good UT for checking whether it is consistent on endianness.